game-core/src/main/java/games/strategy/triplea/delegate/battle/grouptarget/FiringGroup.java

```diff
@@ -0,0 +1,50 @@
+package games.strategy.triplea.delegate.battle.grouptarget;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+
+@Value(staticConstructor = "of")
+public class FiringGroup {
```

Comment:
I think it would be helpful to have a short comment explaining what this represents exactly. How are the units grouped?